Added WaitingTask Chain

Dr15Jones opened the pull request:
#### PR description:

Make it easier to create a chain of WaitingTasks to be run in sequence.

It is now possible to specify the tasks in the order in which they should be run, rather than in reverse order. Works with existing functions that expected to be given a edm::WaitingTaskHolder.

#### PR validation:

New unit test and all framework unit tests pass.

Dr15Jones committed:
Added beginWaitingTaskChain

This is a helper utility to make it easier to create a chain of WaitingTasks. The tasks can be declared in the order in which they should be run.

Dr15Jones committed:
Use beginWaitingTaskChain in EventProcessor

Dr15Jones commented:
NOTE: this is just a refactoring so should have no effect on behavior.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23819

- This PR adds an extra 36KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

- FWCore/Concurrency (**core**)
- FWCore/Framework (**core**)


@makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please review it and eventually sign? Thanks.
@makortel, @wddgit this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy, @perrotta you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Dr15Jones commented:
please test

cmsbuild commented:
-1

**Failed Tests**:   ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16629/summary.html
**COMMIT**: e267bd0dd372a3de23dccd21c9c83f9027c62a40
**CMSSW**: CMSSW_12_0_X_2021-07-08-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34413/16629/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

Dr15Jones committed:
Fixed clang warnings about lambda captures

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23824

- This PR adds an extra 36KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

makortel commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16630/summary.html
**COMMIT**: 3c0440bdec7af5a68da93d32011031490dc43201
**CMSSW**: CMSSW_12_0_X_2021-07-08-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34413/16630/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 5123 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2786260
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2786231
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 37 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-08-2300+bbd486/43944/triggerResults): no differences found

Dr15Jones committed:
Added ifThenNext ability to WaitingTaskChain

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23840

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16658/summary.html
**COMMIT**: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
**CMSSW**: CMSSW_12_0_X_2021-07-09-0800/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34413/16658/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2786260
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2786237
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 37 files compared)
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-09-0800+bbd486/43964/triggerResults): no differences found

Dr15Jones committed:
Changed names of functions for task chains

The naming now follows the convention used by other 3rd party asynchronous interfaces.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23847

- This PR adds an extra 36KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16667/summary.html
**COMMIT**: 0364f8e68ab6d8e086df0847472b4787dc00d545
**CMSSW**: CMSSW_12_0_X_2021-07-09-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34413/16667/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 0
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 0
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.004 KiB( 37 files compared)
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-09-1100+bbd486/43971/triggerResults): no differences found

Dr15Jones committed:
Fold .thenWithException member function into .then

Use SFINAE to detect if the functor wants 1 or 2 arguments.
Moved implementation into _then standalone function to avoid code duplications.

Dr15Jones committed:
Replaced .thenIfExceptionElse method with IfException adaptor

This allows .then to handle this special case as well.

Dr15Jones committed:
Further improvements to names for assembling task chain

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23871

- This PR adds an extra 40KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: c8fdb8294afed112f509874bb614f186527cfc07
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16687/summary.html
**COMMIT**: c8fdb8294afed112f509874bb614f186527cfc07
**CMSSW**: CMSSW_12_0_X_2021-07-11-0000/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/34413/16687/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2787742
 - DQMHistoTests: Total failures: 877
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2786864
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.004 KiB( 37 files compared)
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-11-0000+bbd486/43997/triggerResults): no differences found

makortel commented on FWCore/Concurrency/interface/chain_first.h:
```diff
@@ -0,0 +1,273 @@
+   If a previous task had an exception, only the first functor given to the constructor, iE, will be run and passed the std::exception_ptr const. If there
+   was no exception, then only the functor passed to else_, iF, will be run. If iF has an exception, it will be automatically propagated to the edm::WaitingTaskHolder. */
+    template <typename E>
+    struct IfException {
```
Functionally the `IfException` seems to be used like the `first()`. Even if this would technically violate our naming rules, I'm wondering about
```suggestion
    struct ifException {
```
Then all the pieces of the chaining would start with lower case letter.

Dr15Jones replied:
I could add a function `ifException` which just calls the relevant constructor. That way we do not need to violate our naming conventions.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23874

- This PR adds an extra 12KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: a72849d4b991c50d702a93b3f737671bdb8b421d
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
a couple more changes are coming

Dr15Jones committed:
Added CMS_SA_ALLOW at appropriate locations

Dr15Jones committed:
Declared function ifException and renamed adapter

Dr15Jones commented:
All changes are in

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23875

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: a72849d4b991c50d702a93b3f737671bdb8b421d
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @Dr15Jones can you please check and sign again.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16710/summary.html
**COMMIT**: 2a5734a3e471d937d58dbec94deafeccee1ffe6d
**CMSSW**: CMSSW_12_0_X_2021-07-12-1400/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/34413/16710/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2787742
 - DQMHistoTests: Total failures: 884
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2786857
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 37 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-12-1400+bbd486/44018/triggerResults): no differences found

Dr15Jones committed:
Update descriptions in unit test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23882

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: a72849d4b991c50d702a93b3f737671bdb8b421d
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones committed:
Assemble task chains using operator|

This matches other sequencing APIs used or proposed by the standard.

Dr15Jones committed:
Use new syntax to make task chains

Dr15Jones committed:
Only support operator| or building task chains

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23923

- This PR adds an extra 116KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: a72849d4b991c50d702a93b3f737671bdb8b421d
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
please test

Dr15Jones committed:
Made implementation of then consistent with other parts

-moved functionality of then_ to where it was used
-Have then use has_exception_handler instead of ThenAdaptor

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/23930

- This PR adds an extra 116KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: a72849d4b991c50d702a93b3f737671bdb8b421d
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/src/EventProcessor.cc  modified in PR(s): #33969

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/16784/summary.html
**COMMIT**: 95c92b73ed0a089ec6246ab4209f68d454941ff0
**CMSSW**: CMSSW_12_0_X_2021-07-13-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/34413/16784/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2786302
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2786273
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 37 files compared)
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-13-1100+bbd486/44069/triggerResults): no differences found

makortel commented on FWCore/Concurrency/test/test_catch2_WaitingTaskChain.cc:
```diff
@@ -0,0 +1,515 @@
+                         REQUIRE(false);
+                       }).else_([&count](edm::WaitingTaskHolder h) {
+                   ++count;
+                   REQUIRE(count.load() == 1);
```
Just to note that the indentation is confusing here (I suppose there is little that can be done for it quickly).

Dr15Jones committed:
Fix typo in comment

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34413/24122

- This PR adds an extra 20KB to repository

- Found files with invalid states:
  - FWCore/Concurrency/interface/beginChain.h:
    - Added: 0364f8e68ab6d8e086df0847472b4787dc00d545
    - Modified: 806d33d01f72c2ae9666678cea19c1228db34802, 025f58102735831145d420357795e2de5528c23c
    - Deleted: a72849d4b991c50d702a93b3f737671bdb8b421d
  - FWCore/Concurrency/interface/beginWaitingTaskChain.h:
    - Added: b23cd99751410a0a1819f2298a4ba7acf9f65de8
    - Modified: c55fe89c0f5fe8a9bd2ab52edb49555a29a56cb6
    - Deleted: 0364f8e68ab6d8e086df0847472b4787dc00d545

cmsbuild commented:
Pull request #34413 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

Dr15Jones commented:
please test

it was only a comment change, but what the hey

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/summary.html
**COMMIT**: d24754dbd8b1874adb80d310420dcf5b3b150b14
**CMSSW**: CMSSW_12_0_X_2021-07-20-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34413/17060/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals-INPUT

- [136.732](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.732_RunZeroBias2016B+RunZeroBias2016B+HLTDR2_2016+RECODR2_2016reHLT_ZB_HIPM+HARVESTDR2ZB)```136.732_RunZeroBias2016B+RunZeroBias2016B+HLTDR2_2016+RECODR2_2016reHLT_ZB_HIPM+HARVESTDR2ZB/step2_RunZeroBias2016B+RunZeroBias2016B+HLTDR2_2016+RECODR2_2016reHLT_ZB_HIPM+HARVESTDR2ZB.log```
- [136.726](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.726_RunMuonEG2016B+RunMuonEG2016B+HLTDR2_2016+RECODR2_2016reHLT_skimMuonEG_HIPM+HARVESTDR2)```136.726_RunMuonEG2016B+RunMuonEG2016B+HLTDR2_2016+RECODR2_2016reHLT_skimMuonEG_HIPM+HARVESTDR2/step2_RunMuonEG2016B+RunMuonEG2016B+HLTDR2_2016+RECODR2_2016reHLT_skimMuonEG_HIPM+HARVESTDR2.log```
- [136.721](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.721_RunHLTPhy2016B+RunHLTPhy2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2)```136.721_RunHLTPhy2016B+RunHLTPhy2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2/step2_RunHLTPhy2016B+RunHLTPhy2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.722](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.722_RunDoubleEG2016B+RunDoubleEG2016B+HLTDR2_2016+RECODR2_2016reHLT_skimDoubleEG_HIPM+HARVESTDR2)
- [136.723](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.723_RunDoubleMuon2016B+RunDoubleMuon2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2)
- [136.724](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.724_RunJetHT2016B+RunJetHT2016B+HLTDR2_2016+RECODR2_2016reHLT_skimJetHT_HIPM+HARVESTDR2)
- [136.725](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.725_RunMET2016B+RunMET2016B+HLTDR2_2016+RECODR2_2016reHLT_skimMET_HIPM+HARVESTDR2)
- [136.727](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.727_RunDoubleEGPrpt2016B+RunDoubleEGPrpt2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2)
- [136.728](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.728_RunSingleMuPrpt2016B+RunSingleMuPrpt2016B+HLTDR2_2016+RECODR2_2016reHLT_HIPM+HARVESTDR2)
- [136.729](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.729_RunSingleEl2016B+RunSingleEl2016B+HLTDR2_2016+RECODR2_2016reHLT_L1TEgDQM_HIPM+HARVEST2016_L1TEgDQM)
- [136.73](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.73_RunSingleMu2016B+RunSingleMu2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSingleMu_HIPM+HARVEST2016_L1TMuDQM)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2)
- [136.735](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.735_RunNoBPTX2016B+RunNoBPTX2016B+HLTDR2_2016+RECODR2reHLTAlCaTkCosmics_HIPM+HARVESTDR2)
- [136.734](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.734_RunMuOnia2016B+RunMuOnia2016B+HLTDR2_2016+RECODR2_2016reHLT_skimMuOnia_HIPM+HARVESTDR2)
- [136.72411](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)
- [136.733](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-bbd486/17060/runTheMatrixINPUT-results/136.733_RunCosmics2016B+RunCosmics2016B+RECOCOSDRUN2+ALCACOSDRUN2+HARVESTDCRUN2)
</details>



## Comparison Summary



The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 1222 differences found in the comparisons
 - DQMHistoTests: Total files compared: 39
 - DQMHistoTests: Total histograms compared: 2996268
 - DQMHistoTests: Total failures: 3672
 - DQMHistoTests: Total nulls: 19
 - DQMHistoTests: Total successes: 2992555
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 45.703 KiB( 38 files compared)
 - DQMHistoSizes: changed ( 140.53 ): 44.531 KiB       Hcal/DigiRunHarvesting
 - DQMHistoSizes: changed ( 140.53 ): 1.172 KiB        RPC/DCSInfo
 - Checked 165 log files, 37 edm output root files, 39 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-20-2300+bbd486/44266/triggerResults): no differences found

makortel commented:
Comparison differences are all in 140.53 that had different input file compared to IB. The `RelVals-INPUT` appear to be all
```
A fatal system signal has occurred: external termination request
The following is the call stack containing the origin of the signal.
```
so were the jobs killed because of timeout?

makortel commented:
+1

Earlier tests succeeded fine, the change was only in a comment.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (but tests are reportedly failing). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy, @perrotta (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
merge
